In [1]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

Notas:
* La base de datos de inscritos que se publican en la presente página de internet, es con fecha de corte al 31 de diciembre de 2021.
* Los resultados que se presentan corresponden a la información general que ampara cada título de concesión/asignación y sus bienes públicos inherentes; en este sentido, cabe aclarar que un título de concesión/asignación y sus bienes públicos inherentes, puede contener uno o varios anexos que pueden o no ser del mismo tipo de aprovechamiento.
* Para poder obtener información de los título/constancia se deben especificar los caracteres que integran el número de título/constancia vigente.
* La información contenida en la columna "Autoridad que emite el acto", corresponde a la autoridad administrativa que otorgó o autorizó la más reciente modificación al título de concesión, asignación, permiso o registro de que se trate.
* El dato de Cuenca Hidrológica se presenta con el enfoque adaptativo que permitirá brindar mayor certeza a los usuarios respecto de la ubicación del lugar donde se lleva a cabo la explotación uso o aprovechamiento de las aguas, así como mejorar la administración y gestión de los recursos hídricos, mismo que es utilizado para realizar los cálculos de la disponibilidad media anual de las aguas superficiales.


In [2]:
repdaTyPBasico = pd.read_csv('./ReportRepdaBasico.csv', sep = ",")

In [3]:
repdaTyPBasico.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 529 entries, 0 to 528
Data columns (total 14 columns):
 #   Column                                              Non-Null Count  Dtype  
---  ------                                              --------------  -----  
 0   Titular                                             529 non-null    object 
 1   Título                                              529 non-null    object 
 2   Uso                                                 529 non-null    object 
 3   Autoridad que emite el acto                         529 non-null    object 
 4   Fecha de registro                                   529 non-null    object 
 5   Volumen de extracción de aguas nacionales (m3/a?o)  529 non-null    float64
 6   Número de anexos de aguas superficiales             529 non-null    int64  
 7   Volumen de aguas superficiales (m3/a?o):            529 non-null    float64
 8   Número de anexos de aguas subterráneas              529 non-null    int64  
 9  

In [5]:
repdaTyPBasico['Fecha de registro'] = repdaTyPBasico['Fecha de registro'].apply(pd.to_datetime)

In [6]:
repdaTyPBasico

Titular               Título  \
0                           JOSE OTONIEL GOMEZ SANCHEZ               843899   
1    ORGANISMO OPERADOR DE AGUA POTABLE, ALCANTARIL...               842972   
2                       CHRISTIAN JANETH GUILLEN AYALA               816259   
3                                ARTURO NARANJO PULIDO               839107   
4    RESIDENTES Y DESCONCENTRADOS DE LA SECRETARIA ...               837890   
..                                                 ...                  ...   
524                              ARTURO NAVA VELAZQUEZ  4MCH100608/12LCGE94   
525                    CASA DE LA MUJER CENOPISTA A.C.  4MCH100581/12HMGE94   
526  CONSORCIO INMOBILIARIO DE MORELIA S.A. DE C.V....  4MCH100584/12HMGE94   
527      TURBINAS Y EQUIPOS INDUSTRIALES S.A. DE C. V.  4MCH100002/12ERGE94   
528                   COMISION FEDERAL DE ELECTRICIDAD  4MCH100208/12JBSG94   

                                 Uso  \
0                           AGRICOLA   
1                     PUBLICO URBANO   
2                          SERVICIOS   
3                           AGRICOLA   
4                           AGRICOLA   
..                               ...   
524                            OTROS   
525                   PUBLICO URBANO   
526                   PUBLICO URBANO   
527                        SERVICIOS   
528  GENERACION DE ENERGIA ELECTRICA   

                         Autoridad que emite el acto Fecha de registro  \
0                          DIRECCIÓN LOCAL MICHOACÁN        2021-12-16   
1                          DIRECCIÓN LOCAL MICHOACÁN        2021-10-27   
2                          DIRECCIÓN LOCAL MICHOACÁN        2021-07-09   
3                          DIRECCIÓN LOCAL MICHOACÁN        2021-03-09   
4                          DIRECCIÓN LOCAL MICHOACÁN        2021-08-23   
..                                               ...               ...   
524                        DIRECCIÓN LOCAL MICHOACÁN        1994-08-17   
525                        DIRECCIÓN LOCAL MICHOACÁN        1994-11-08   
526                        DIRECCIÓN LOCAL MICHOACÁN        1994-09-08   
527                        DIRECCIÓN LOCAL MICHOACÁN        1994-07-07   
528  SUBDIRECCIÓN GENERAL DE ADMINISTRACIÓN DEL AGUA        1994-05-13   

     Volumen de extracción de aguas nacionales (m3/a?o)  \
0                                              30000.0    
1                                             251028.7    
2                                                  0.0    
3                                             110000.0    
4                                              10000.0    
..                                                 ...    
524                                                0.0    
525                                           314000.0    
526                                                0.0    
527                                                0.0    
528                                         17125000.0    

     Número de anexos de aguas superficiales  \
0                                          0   
1                                          0   
2                                          0   
3                                          0   
4                                          0   
..                                       ...   
524                                        0   
525                                        0   
526                                        0   
527                                        0   
528                                        1   

     Volumen de aguas superficiales (m3/a?o):  \
0                                         0.0   
1                                         0.0   
2                                         0.0   
3                                         0.0   
4                                         0.0   
..                                        ...   
524                                       0.0   
525                                       0.0   
5

In [7]:
repdaTyPBasico.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 529 entries, 0 to 528
Data columns (total 14 columns):
 #   Column                                              Non-Null Count  Dtype         
---  ------                                              --------------  -----         
 0   Titular                                             529 non-null    object        
 1   Título                                              529 non-null    object        
 2   Uso                                                 529 non-null    object        
 3   Autoridad que emite el acto                         529 non-null    object        
 4   Fecha de registro                                   529 non-null    datetime64[ns]
 5   Volumen de extracción de aguas nacionales (m3/a?o)  529 non-null    float64       
 6   Número de anexos de aguas superficiales             529 non-null    int64         
 7   Volumen de aguas superficiales (m3/a?o):            529 non-null    float64       
 8   Número de 

In [8]:
repdaTyPBasico['Uso'].value_counts()

SERVICIOS                          190
AGRICOLA                           130
PUBLICO URBANO                      90
INDUSTRIAL                          38
DIFERENTES USOS                     34
OTROS                               20
DOMESTICO                           18
PECUARIO                             6
ACUACULTURA                          2
GENERACION DE ENERGIA ELECTRICA      1
Name: Uso, dtype: int64

Búsqueda normal -> 529 registros </br>
Búsqueda avanzada -> 534 registros

¿Cómo determinan el Tipo de aprovechamiento?

No se puede leer directamente, toca cambiarle el formato a html, copíar y pegar en una hija cálculo y de ahí exportar

In [9]:
repTaproch1 = pd.read_csv('./ReportRepdaApr1.csv')
repTaproch2 = pd.read_csv('./ReportRepdaApr2.csv')
repTaproch3 = pd.read_csv('./ReportRepdaApr3.csv')
repTaproch4 = pd.read_csv('./ReportRepdaApr4.csv')

In [10]:
repTaproch1['Tipo Aprovechamiento'] = 'Superficiales'
repTaproch2['Tipo Aprovechamiento'] = 'Subterráneos'
repTaproch3['Tipo Aprovechamiento'] = 'Descargas Residuales'
repTaproch4['Tipo Aprovechamiento'] = 'Zonas Federales'

In [11]:
repdaTyPAvanzado = pd.concat([repTaproch1,repTaproch2,repTaproch3,repTaproch4])

In [16]:
repdaTyPAvanzado.to_csv('./ReportRepdaAvanzado.csv', index=False)

In [18]:
repdaTyPAvanzado = pd.read_csv('./ReportRepdaAvanzado.csv')

In [19]:
repdaTyPAvanzado['Fecha de registro'] = repdaTyPAvanzado['Fecha de registro'].apply(pd.to_datetime)


In [20]:
repdaTyPAvanzado.head()

Titular                Título  \
0                UNIVERSIDAD LATINA DE AMERICA, A.C.                830044   
1  ORGANISMO OPERADOR DE AGUA POTABLE, ALCANTARIL...  08MCH160778/12HODL18   
2                                ALFONSO MENDEZ CEJA  04MCH159760/18ADDL18   
3                     J. ASCENCION MONDRAGON BENITIS  08MCH159647/12ADDL17   
4                         ROELL RODRIGO TELLO MOLINA  08MCH159652/12ADDL17   

              Uso Autoridad que emite el acto Fecha de registro  \
0       SERVICIOS   DIRECCIÓN LOCAL MICHOACÁN        2020-09-18   
1  PUBLICO URBANO   DIRECCIÓN LOCAL MICHOACÁN        2018-11-10   
2        AGRICOLA   DIRECCIÓN LOCAL MICHOACÁN        2018-02-08   
3        AGRICOLA   DIRECCIÓN LOCAL MICHOACÁN        2018-02-08   
4        AGRICOLA   DIRECCIÓN LOCAL MICHOACÁN        2018-02-08   

   Volumen de extracción de aguas nacionales (m3/a?o)  \
0                                            16610.0    
1                                           189216.0    
2                                            12000.0    
3                                            12500.0    
4                                            10000.0    

   Número de anexos de aguas superficiales  \
0                                        1   
1                                        1   
2                                        1   
3                                        1   
4                                        1   

   Volumen de aguas superficiales (m3/a?o):  \
0                                   16610.0   
1                                  189216.0   
2                                   12000.0   
3                                   12500.0   
4                                   10000.0   

   Número de anexos de aguas subterráneas  \
0                                     0.0   
1                                     0.0   
2                                     0.0   
3                                     0.0   
4                                     0.0   

   Volumen de aguas subterráneas (m3/a?o):  Número de anexos de descarga:  \
0                                      0.0                            0.0   
1                                      0.0                            0.0   
2                                      0.0                            0.0   
3                                      0.0                            0.0   
4                                      0.0                            0.0   

   Volumen de descarga (m3/día):  Número de anexos de zonas federales:  \
0                            0.0                                   0.0   
1                            0.0                                   0.0   
2                            0.0                                   0.0   
3                            0.0                                   0.0   
4                            0.0                                   0.0   

   Superficie (m2): Tipo Aprovechamiento  
0               0.0        Superficiales  
1               0.0        Superficiales  
2               0.0        Superficiales  
3               0.0        Superficiales  
4               0.0        Superficiales

In [21]:
repdaTyPAvanzado.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 534 entries, 0 to 533
Data columns (total 15 columns):
 #   Column                                              Non-Null Count  Dtype         
---  ------                                              --------------  -----         
 0   Titular                                             534 non-null    object        
 1   Título                                              534 non-null    object        
 2   Uso                                                 534 non-null    object        
 3   Autoridad que emite el acto                         534 non-null    object        
 4   Fecha de registro                                   534 non-null    datetime64[ns]
 5   Volumen de extracción de aguas nacionales (m3/a?o)  534 non-null    float64       
 6   Número de anexos de aguas superficiales             534 non-null    int64         
 7   Volumen de aguas superficiales (m3/a?o):            534 non-null    float64       
 8   Número de 

**No hay registros para** 
+ Títulos con permisos de extracción de materiales pétreos
+ Títulos de aguas nacionales otorgados a distritos de riego
+ Registro de obras en zonas de libre alumbramiento

In [23]:
repdaTyPBasico.columns, repdaTyPAvanzado.columns

(Index(['Titular', 'Título', 'Uso', 'Autoridad que emite el acto',
        'Fecha de registro',
        'Volumen de extracción de aguas nacionales (m3/a?o)',
        'Número de anexos de aguas superficiales',
        'Volumen de aguas superficiales (m3/a?o):',
        'Número de anexos de aguas subterráneas',
        'Volumen de aguas subterráneas (m3/a?o):',
        'Número de anexos de descarga:', 'Volumen de descarga (m3/día):',
        'Número de anexos de zonas federales:', 'Superficie (m2):'],
       dtype='object'),
 Index(['Titular', 'Título', 'Uso', 'Autoridad que emite el acto',
        'Fecha de registro',
        'Volumen de extracción de aguas nacionales (m3/a?o)',
        'Número de anexos de aguas superficiales',
        'Volumen de aguas superficiales (m3/a?o):',
        'Número de anexos de aguas subterráneas',
        'Volumen de aguas subterráneas (m3/a?o):',
        'Número de anexos de descarga:', 'Volumen de descarga (m3/día):',
        'Número de anexos de zonas 

In [26]:
comunes = np.intersect1d(repdaTyPBasico['Título'], repdaTyPAvanzado['Título'])

In [27]:
repdaTyPAvanzado[~repdaTyPAvanzado['Título'].isin(comunes)]

Empty DataFrame
Columns: [Titular, Título, Uso, Autoridad que emite el acto, Fecha de registro, Volumen de extracción de aguas nacionales (m3/a?o), Número de anexos de aguas superficiales, Volumen de aguas superficiales (m3/a?o):, Número de anexos de aguas subterráneas, Volumen de aguas subterráneas (m3/a?o):, Número de anexos de descarga:, Volumen de descarga (m3/día):, Número de anexos de zonas federales:, Superficie (m2):, Tipo Aprovechamiento]
Index: []